app/models/content.rb

```diff
@@ -28,9 +28,6 @@ def update_content_attributes(content)
   end
 
   def replace_resources!
-    episode.with_lock do
-      # delete enclosures in the same episode created before this
-      episode.all_contents.where("position = ? AND created_at < ? AND id != ?", position, created_at, id).destroy_all
-    end
+    Content.where(episode_id: episode_id, position: position).where.not(id: id).touch_all(:replaced_at, :deleted_at)
```

Comment:
Okay here's the fun part...

For all the audio/image files, there's an `after_create` hook.  It finds any other files we're replacing, and marks them both as `deleted_at` and `replaced_at`.  Which allows us to distinguish these "replaced" files vs just deleted ones in the `complete_or_replaced` scope.

Reply:
One thing that's not clear is when validation / transcoding could happen.  Is that via the `complete` enum?  Or maybe we are already downstream of the porter tasks.